dot/network/transaction.go

```diff
@@ -119,36 +120,54 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
-func (s *Service) createBatchMessageHandler(txnBatch chan *BatchMessage) NotificationsMessageBatchHandler {
-	return func(peer peer.ID, msg NotificationsMessage) (msgs []*BatchMessage, err error) {
+func (s *Service) createBatchMessageHandler(txnBatchCh chan *BatchMessage) NotificationsMessageBatchHandler {
+	go func() {
+		protocolID := s.host.protocolID + transactionsID
+		ticker := time.NewTicker(s.cfg.SlotDuration)
+		defer ticker.Stop()
+
+		for {
+		out:
+			select {
+			case <-s.ctx.Done():
+				return
+			case <-ticker.C:
+				timeOut := time.NewTimer(s.cfg.SlotDuration / 3)
+				for {
+					select {
+					case <-timeOut.C:
+						break out
+					case txnMsg := <-txnBatchCh:
```

Comment:
I think that will cause a memory leak. We should stop the timer and drain its channel

```go
if !timer.Stop() {
	<-timer.C
}
```

Reply:
@qdm12 Can you provide more details why there will be a memory leak?
We only exit the loop when we hit the timeout. So every exit from the loop drains the timer channel.

Reply:
My bad, I got confused with the goto and label!